connectors/sources/sharepoint_online.py

```diff
@@ -1732,13 +1797,29 @@ def _get_id(permissions, identity):
 
             user_id = _get_id(granted_to_v2, "user")
             group_id = _get_id(granted_to_v2, "group")
+            site_group_id = _get_id(granted_to_v2, "siteGroup")
+            site_user_email = _get_email(granted_to_v2, "siteUser")
```

Comment:
Does anything add items to the user ALC list with `siteGroup` and `siteUser` prefixes? It looks like the logic gated under
```
if is_sharepoint_group(user):
```
1. doesn't add a group permission/id at all
2. doesn't prefix the user ids with `siteUser`